acq4/drivers/MaiTai/maitai.py

```diff
@@ -0,0 +1,280 @@
+        #    raise Exception("New pump laser output power is higher than the last commanded. Last command : %s ; New :  %s" % (lastCommandedPower, ppower) )
+        #else:
+        self['PLASer:POWer'] = float(ppower)
+    
+    def getShutter(self):
+        """Return True if the shutter is open."""
+        return bool(int(self['SHUTter?']))
+    
+    def setShutter(self, val):
+        """Open (True) or close (False) the shutter"""
+        while self.getShutter() != val:
+            self['SHUTter'] = (1 if val else 0)
+        if val:
+            print 'Shutter OPEN'
+        else:
+            print 'Shutter CLOSED'
+        
+    def getPumpMode(self):
+        """ returns pump mode of the laser """
+        crypticMode = self['MODE?']
+        return self.modeNames[crypticMode]
+    
+    def setPumpMode(self, mode):
+        """ sets the pump mode of the laser """
+        oldMode = self.getPumpMode()
+        for k in self.modeNames :
+            if mode == self.modeNames[k] :
+                self['MODE'] = k
+        newMode = self.getPumpMode()
+        print 'changedMode : ', oldMode, newMode
+        
+    def getSystemIdentification(self):
+        """Return a system identification string that contains 4 fields separated by commas."""
+        return self['*IDN?']
+    
+    def checkPulsing(self):
+        """Return True if the laser is pulsing."""
+        status = int(self['*STB?'])  #Returns the product status byte.
+        return self.is_set(status,1)
+    
+    def is_set(self,x, n):
+        """ checks whether n^th bit is set"""
+        return (x & 2**n != 0)
+
+    def turnLaserOn(self):
+        
+        while True:
+            warmedUP = self.convertToFloat(self['READ:PCTWarmedup?'])
+            if warmedUP == 100.:
+                break
+            else:
+                print ('System warming up. Currently at %f ' % warmedUP)
+                time.sleep(self.waitTime)
+
+        self.write('ON\r')
+        time.sleep(self.waitTime)
+        print 'LASER IS ON'
+        
+    def turnLaserOff(self):
+        if self.getShutter():
+            self.setShutter(False)
+        self.write('OFF\r')
+        time.sleep(self.waitTime)
+        print 'LASER IS OFF'
+    
+    def __getitem__(self, arg):  ## request a single value from the laser
+        #print "write", arg
+        self.write("%s\r" % arg)
+        ret = self.readPacket()
+        #print "   return:", ret
+        return ret
+        
+    def __setitem__(self, arg, val):  ## set a single value on the laser
+        #print "write", arg, val
+        self.write("%s %s\r" % (arg,str(val)))
+        #ret = self.readPacket()
+        #print "   return:", ret
+        #return ret
+
+    def clearBuffer(self):
+        d = self.read()
+        time.sleep(0.1)
+        d += self.read()
+        if len(d) > 0:
+            print "Sutter MP285: Warning: tossed data ", repr(d)
+        return d
+    
+    def read(self):
```

Comment:
If you use SerialDevice correctly, then it probably won't be necessary to reimplement read / write / close / readPacket.  (and you can use readUntil instead of readPacket)